workers/unity/Packages/com.improbable.gdk.core/Utility/PlayerLoopUtils.cs

```diff
@@ -201,5 +201,39 @@ public static void PrintPlayerLoop()
                 }
             }
         }
+
+        public static void RemoveFromPlayerLoop(World world)
+        {
+            var playerLoop = ScriptBehaviourUpdateOrder.CurrentPlayerLoop;
+            if (playerLoop.subSystemList == null)
+            {
+                Debug.LogWarning("Cannot remove a world from default PlayerLoop.");
+                return;
+            }
+
+            //Reflection to get world from PlayerLoopSystem
+            var wrapperType =
+                typeof(ScriptBehaviourUpdateOrder).Assembly.GetType(
+                    "Unity.Entities.ScriptBehaviourUpdateOrder+DummyDelegateWrapper");
+            var systemField = wrapperType.GetField("m_System", BindingFlags.NonPublic | BindingFlags.Instance);
+
+            for (var i = 0; i < playerLoop.subSystemList.Length; ++i)
+            {
+                ref var subSystem = ref playerLoop.subSystemList[i];
+                subSystem.subSystemList = subSystem.subSystemList.Where(s =>
+                {
+                    if (s.updateDelegate != null && s.updateDelegate.Target.GetType() == wrapperType)
+                    {
+                        var system = systemField.GetValue(s.updateDelegate.Target) as ComponentSystemBase;
```

Comment:
here are way too many system fields, and then systems that store lists of systems. are they all systems?

Reply:
They are all systems

Reply:
is there a way to name all these things slightly more sensible? 
just having s, subsystems and systems as names makes it a bit more difficult to figure out which system we talk about